include/platform/mir/graphics/program_factory.h

```diff
@@ -41,16 +41,21 @@ class ProgramFactory
     /**
      * Compile and link a fragment-shader fragment into a full shader Program
      *
-     * \param extension_fragment [in]    Fragment to include necessary extensions
+     * \note    As the id needs to be globally unique, we suggest using the address of a local
+     *          static variable as the id.
```

Comment:
Does that make it globally unique? A local static variable will have the same address in multiple invocations.

Reply:
Hm. I've clearly communicated badly.

The `id` needs to be the same for each invocation of `compile_fragment_shader` for the same shader, and different for each different shader.

It doesn't need to (and, in fact, needs *not* to be) be unique for each call.